### PR TITLE
[hail] Remove {MatrixIR, TableIR}.rvdType

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by cseed** [comment]
1. I think you can remove `rvRowType` from `MatrixType` now, which is awesome.

2. Can you also remove TableIR.rvdType?  It doesn't seem to be used substantively.

**Comment by tpoterba** [comment]
> I think you can remove rvRowType from MatrixType now, which is awesome.

Yes! It's a bit of work to pull it out, but that's coming up next.

**Comment by tpoterba** [comment]
> Can you also remove TableIR.rvdType? It doesn't seem to be used substantively.

Yeah, I think so. I might want to add back a physicalKey field though?

**Comment by cseed** [comment]
Do you need the physical key statically?  The RVD will always have the physical key.

**Comment by tpoterba** [comment]
> Do you need the physical key statically? The RVD will always have the physical key.

well, here's an example:
https://github.com/hail-is/hail/pull/6231/files#diff-77a7afce9203811118cd6aac172a8a76L497

I want to add that back in when we have a way to define rules that can only operate on lowered IR, for instance.

**Comment by cseed** [comment]
Good example.  You can store physical key as a field, but you can also think of it as the result of an analysis, like prune and nesting depth, which you can compute in the optimizer as needed.  Just another organizational option.

**Comment by tpoterba** [comment]
yeah, absolutely. not sure which is preferable.

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala [hunk 1]
```diff
@@ -1099,6 +1105,7 @@ case class MatrixRowsTable(child: MatrixIR) extends TableIR {
   }
 
   val typ: TableType = child.typ.rowsTableType
+  def rvdType: RVDType = throw new NotImplementedError()
```
Comment (cseed) [inline hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: Can you get away with this?  This screams (1) latent bug or (2) rvdType is actually unused.

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: yeah, I'll see if I can peel it off table too

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: yes, rvdType gone from table too!

Comment (cseed) [reply hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: Woot!

**Pull request merged**